### PR TITLE
fix: automatic db creation

### PR DESCRIPTION
**Pull request by luzzif**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,7 +34,13 @@ format is the same as the `.config.example.yaml` file at the root of the repo.
...
 values that you want. By default, the data regarding contract addresses and
 deployment blocks should be correct, so you'd only need to change the other
-data.
+configs.
+
+> [!IMPORTANT]  
+> The program will automatically create the database as specified in the
+> Postgres connection string in the config (key `db_connection_string`), but it
+> will do so only if the specified user and password combination are correct, so
+> double check the Docker Compose condifuration to set them right.
```
Comment (guerrap) [inline README.md]: Typo here `condifuration`.

File: src/db.rs [hunk 2]
```diff
@@ -1,19 +1,69 @@
...
+            tracing::info!("creating database {}", database);
+
+            // connect to "postgres" database
+            let mut pg_db_parsed_url = parsed_url.clone();
+            pg_db_parsed_url.set_path("postgres");
+            let mut pg_db_connection = PgConnection::establish(pg_db_parsed_url.as_str())
+                .context("could not connect to admin postgres database")?;
+
+            // create database
+            diesel::sql_query(format!("CREATE DATABASE \"{}\";", database))
```
Comment (guerrap) [inline src/db.rs:48]: I guess it would fail to create the database if it already exists right?

Comment (luzzif) [reply src/db.rs:48]: Exactly, it would just fail and stop

File: README.md (hunk 1 above)
Comment (luzzif) [reply README.md]: Addressed in b3faae76f5cd1b7c561ef963da7d57102b057b6b

**Pull request merged**